### ISSUE TITLE
Functionality Issues?

### ISSUE THREAD

**Issue by SurpriseSnek** [AI-Insider/dcgan-facegenerator#1]
While I'm no professional, and this is certainly an amazing project, I've loaded it up into python and I keep running into a lot of issues. Mainly how it seems rather reluctant to start doing anything out of the box. I've managed to setup everything and even train some weights, but the Generator's loss doesn't seem to go down, and in fact seems to steadily climb upwards. It maxes out around 16.118, and won't go higher. It also only produces fully noisy images. I've run it for 10,000+ epochs and it still encounter's these two main issues. While I didn't run it on faces, and this could be the issue, it still feeds in the images and takes a look at them. Any help would be appreciated, and if you need any information, I'll be glad to give it. Thanks

**Comment by jordan-bird** [comment]
There could be a great many reasons as to why this is happening but the most obvious one, and the reason I've hit 16.118 loss for a dataset of my own 10k images before is because the discriminator is too strong and the generator is too weak. Once the gradients are lost then no more learning will happen, whether it be 10k or 10 million epochs, those losses won't change any more.

I've been playing around with hyperparameters and so far I've come up with this:
https://github.com/jordan-bird/art-DCGAN-Keras

If I increase the kernel size of the generator then the whole thing completely collapses. Right now I have the above training on a dataset and it's currently at epoch 200 without any sign of collapse. 

Try copy and pasting my network topologies into your code and see what happens, and if they don't work here are some suggestions:

1. Dropout layers
2. Leaky ReLu
3. Batch normalisation
4. Fewer Discriminator layers
5. Convolutions within the generator

Jason Brownlee has compiled some of the scientific state-of-the-art suggestions to improve GANs:
https://machinelearningmastery.com/how-to-code-generative-adversarial-network-hacks/

Good luck!
